openvm/src/lib.rs

```diff
@@ -400,51 +401,39 @@ pub fn prove(
-            let proof =
-                sdk.generate_app_proof(app_pk.clone(), app_committed_exe.clone(), inputs.clone())?;
-            tracing::info!("Proof generation done.");
-
-            tracing::info!(
-                "Public values: {:?}",
-                proof.user_public_values.public_values
-            );
+        // Generate a proof
+        tracing::info!("Generating app proof...");
+        let app_proof =
+            sdk.generate_app_proof(app_pk.clone(), app_committed_exe.clone(), inputs.clone())?;
+        tracing::info!("App proof generation done.");
+
+        tracing::info!(
+            "Public values: {:?}",
+            app_proof.user_public_values.public_values
+        );
+
+        // Verify
+        let app_vk = app_pk.get_app_vk();
+        sdk.verify_app_proof(&app_vk, &app_proof)?;
+        tracing::info!("App proof verification done.");
 
-            // Verify
-            let app_vk = app_pk.get_app_vk();
-            sdk.verify_app_proof(&app_vk, &proof)?;
-            tracing::info!("Proof verification done.");
-        } else {
+        if recursion {
             // Generate the aggregation proving key
-            const DEFAULT_PARAMS_DIR: &str = concat!(env!("HOME"), "/.openvm/params/");
-            let halo2_params_reader = CacheHalo2ParamsReader::new(DEFAULT_PARAMS_DIR);
-            let agg_config = AggConfig::default();
             tracing::info!("Generating aggregation proving key...");
-            let agg_pk = sdk.agg_keygen(
-                agg_config,
-                &halo2_params_reader,
-                &DefaultStaticVerifierPvHandler,
-            )?;
-
-            tracing::info!("Generating SNARK verifier...");
-            // Generate the SNARK verifier smart contract
-            let verifier = sdk.generate_halo2_verifier_solidity(&halo2_params_reader, &agg_pk)?;
-
-            tracing::info!("Generating EVM proof...");
-            // Generate an EVM proof
-            let proof = sdk.generate_evm_proof(
-                &halo2_params_reader,
-                app_pk,
-                app_committed_exe,
-                agg_pk,
-                inputs,
-            )?;
-
-            tracing::info!("Verifying EVM proof...");
-            // Verify the EVM proof
-            sdk.verify_evm_halo2_proof(&verifier, proof)?;
+            let (agg_stark_pk, _) =
+                AggStarkProvingKey::dummy_proof_and_keygen(AggStarkConfig::default());
+
+            tracing::info!("Generating aggregation proof...");
+
+            let agg_prover = AggStarkProver::<BabyBearPoseidon2Engine>::new(
+                agg_stark_pk,
+                app_pk.leaf_committed_exe.clone(),
+                *sdk.agg_tree_config(),
+            );
+            // Note that this proof is not verified. We assume that any valid app proof
+            // (verified above) also leads to a valid aggregation proof.
+            // If this was not the case, it would be a completeness bug in OpenVM.
```

Comment:
I didn't find an easy way to verify the proof, and I'm not sure a native verifier is implemented. But as mentioned here, I don't see how we should be causing invalid proofs if the app proofs are valid.